internal/bminventory/inventory.go

```diff
@@ -5263,7 +5263,7 @@ func (b *bareMetalInventory) V2UpdateHostInstallProgress(ctx context.Context, pa
 			return installer.NewV2UpdateHostInstallProgressNotFound().
 				WithPayload(common.GenerateError(http.StatusNotFound, err))
 		}
-		return installer.NewV2UpdateHostInstallProgressInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+		return installer.NewV2UpdateHostInstallProgressConflict().WithPayload(common.GenerateError(http.StatusConflict, err))
```

Comment:
I think this is too wide. There are a lot of things that could go wrong in `V2UpdateHostInstallProgressInternal` that are worthy of 500 errors rather than 409.

I'm hoping that you can find the state incompatibility issues and return an [`ApiErrorResponse`](https://github.com/openshift/assisted-service/blob/5e951a16d3446d5c0188f3d4454687757537b7d5/internal/common/error_utils.go#L52) up from those then change this to use [`common.GenerateErrorResponder`](https://github.com/openshift/assisted-service/blob/5e951a16d3446d5c0188f3d4454687757537b7d5/internal/common/error_utils.go#L118-L133) which will either return the error it was given or create a new 500 error.

Reply:
Gotcha, thanks for the pointers! I went ahead and modified the error responses and used the common.GenerateErrorResponder. How does it look now?